src/Altinn.AccessManagement.Core/Services/DelegationsService.cs

```diff
@@ -20,80 +23,120 @@ public class DelegationsService : IDelegationsService
+                throw new ValidationException($"Maskinporten schema delegation can only be used to delegate maskinporten schemas. Invalid resource: {resourceRegistryId}. Invalid resource type: {resource.ResourceType}");
+            }
 
-                if (directMainUnitDelegations.Any())
-                {
-                    delegations.AddRange(directMainUnitDelegations);
-                }
+            // Verify To recipient of delegation
+            if (!DelegationHelper.TryGetPartyIdFromAttributeMatch(delegation.To, out int partyId))
+            {
+                throw new ValidationException($"Maskinporten schema delegation currently only support delegation To a PartyId (using {AltinnXacmlConstants.MatchAttributeIdentifiers.PartyAttribute} attribute id). Invalid value: {delegation.To.FirstOrDefault()?.Id}");
             }
 
-            // 3. Direct party delegations to keyrole units
-            List<int> keyrolePartyIds = await _contextRetrievalService.GetKeyRolePartyIds(subjectUserId);
-            if (keyrolePartyIds.Any())
+            Party toParty = await _contextRetrievalService.GetPartyAsync(partyId);
+            if (toParty == null || toParty.PartyTypeName != PartyType.Organisation)
+            {
+                throw new ValidationException($"Maskinporten schema delegation can only be delegated To a valid PartyId belonging to an organization. Invalid value: {partyId}");
+            }
+
+            // Verify authenticated users delegable rights
+            RightsQuery rightsQuery = new RightsQuery
+            {
+                To = new List<AttributeMatch> { new AttributeMatch { Id = AltinnXacmlConstants.MatchAttributeIdentifiers.UserAttribute, Value = delegatingUserId.ToString() } },
+                From = new List<AttributeMatch> { new AttributeMatch { Id = AltinnXacmlConstants.MatchAttributeIdentifiers.PartyAttribute, Value = from } },
+                Resource = right.Resource
+            };
+            List<Right> usersDelegableRights = await _pip.GetRights(rightsQuery, getDelegableRights: true);
+            if (usersDelegableRights.IsNullOrEmpty())
             {
-                List<DelegationChange> keyRoleDelegations = resourceMatchType == ResourceAttributeMatchType.AltinnAppId ?
-                    await _delegationRepository.GetAllCurrentAppDelegationChanges(offeredByPartyIds, resourceIds, coveredByPartyIds: keyrolePartyIds) :
-                    await _delegationRepository.GetAllCurrentResourceRegistryDelegationChanges(offeredByPartyIds, resourceIds, coveredByPartyIds: keyrolePartyIds);
+                throw new ValidationException($"Authenticated user does not have any delegable rights for the resource: {resourceRegistryId}");
+            }
+
+            if (usersDelegableRights.Any(r => !r.CanDelegate.Value)) 
+            {
+                // ToDo: include SecurityLevelInfo on delegable rights
```

Comment:
Resolve todo. Info about security level is implemented and must be checked here

Reply:
implemented validation check of the minimum authentication level requirement of the resource